### DISCUSSION TITLE
Order of execution of events in the QMenu component

### DISCUSSION THREAD

**Discussion by KINGER404** [quasarframework/quasar#13579] [General - Components / Directives / etc] [answered]
When opening a menu, @update:model-value fires first, and then @before-show, similarly to closing

https://codepen.io/kinger404/pen/LYQmYbj?editors=101

**Comment by metalsadman** [comment, marked as answer]
it's correct `@before-show -> function(evt)
Description
Emitted when component triggers show() but before it finishes doing it
`

**Reply by KINGER404** [reply]
I thought so, I just decided to clarify. Thank you.
